### DISCUSSION TITLE
Not possible to change logfile location for `nextcloud-aio-nextcloud`?

### DISCUSSION THREAD

**Discussion by nreymundo** [nextcloud/all-in-one#2194] [Questions] [answered]
Hello!, 

I'm trying to migrate from another install of Nextcloud (through docker) to one with the all-in-one setup for ease of maintenance. Everything is going great so far and have pretty much all I need configured but I'm having issues trying to change the location of the `nextcloud.log` file. 

The reason why I want to change it is so that I can have CrowdSec consume the logs and take security measures network-wide in case of repeated invalid logins. 

The default line in the `config.php` file is: 
```
'logfile' => '/var/www/html/data/nextcloud.log',
```

And I tried changing it to
```
'logfile' => '/mnt/ncdata/nextcloud.log',
```

But then the moment the container restarts it gets overwritten back to the previous value. 

Is there any way to do what I'm attempting or just not possible with nextcloud-aio?. 
Thanks!.

**Comment by szaimen** [comment, marked as answer]
Hi, I think you should be able to use the samr location as advised in https://github.com/nextcloud/all-in-one#fail2ban

**Reply by nreymundo** [reply]
Awesome! That totally did the trick. 

Sorry, should have thought about looking up options for fail2ban since they both work so similarly.

**Comment by caalca** [comment]
I want to protect nextcloud with Crodwsec that is installed in the host, and it is connected to take action with nginx proxy manager that is also the host and it specified as the reverse proxy for Nextcloud-AIO. 

It is necessary to access nextcloud logs from the host, i have change the location of the log to `/mnt/ncdata/` (where is mounted the external disk). But nextcloud changes to default location `/var/www/html/data/nextcloud.log`  at restart of docker container.

It would be nice to add an enviroment variable that let us change the log file location. 

A quick quick workaround I have made is use crontab that executes this command every minute and update config.php to the externa disk location.
```
sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set --type string --value /mnt/ncdata/nextcloud.log logfile
```

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#fail2ban, there you can find where the log is located by default
